source/common/upstream/http_status_checker.cc

```diff
@@ -0,0 +1,61 @@
+#include "common/upstream/http_status_checker.h"
+
+#include <vector>
+
+#include "envoy/api/v2/core/health_check.pb.h"
+
+#include "common/access_log/access_log_formatter.h"
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Upstream {
+
+HttpStatusCheckerPtr HttpStatusChecker::configure(
+    const Protobuf::RepeatedPtrField<envoy::type::Int64Range>& expected_statuses,
+    uint64_t default_expected_status) {
+  HttpStatusCheckerPtr httpStatusChecker(new HttpStatusChecker());
+
+  for (const auto& status_range : expected_statuses) {
+    auto start = status_range.start();
+    auto end = status_range.end();
+
+    if (start >= end) {
+      throw EnvoyException(fmt::format(
+          "Invalid http status range: expecting start < end, but found start={} and end={}", start,
+          end));
+    }
+
+    if (start < 100) {
+      throw EnvoyException(fmt::format(
+          "Invalid http status range: expecting start >= 100, but found start={}", start));
+    }
+
+    if (end > 600) {
+      throw EnvoyException(
+          fmt::format("Invalid http status range: expecting end <= 600, but found end={}", end));
+    }
+
+    httpStatusChecker->ranges_.emplace_back(
+        std::make_pair(static_cast<uint64_t>(start), static_cast<uint64_t>(end)));
+  }
+
+  if (httpStatusChecker->ranges_.empty()) {
```

Comment:
Should we keep `200` irrespective of the input ranges?

Reply:
In my opinion it is better to give users full control of accepted responses. Users will probably just extend the range using [200, 300) or [200, 400), so 200s will be included explicitly. I don't know why someone would need to disable 200s, but if they wanted, they could.

Reply:
Can we explictly mention this behavior in the proto (or elsewhere)? Thanks.

Reply:
I clarified comment in proto file, it should be clear now.